pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -615,7 +618,47 @@ private static Expression toExpression(SqlNode node) {
             funcExpr.getFunctionCall().addToOperands(toExpression(child));
           }
         }
+
+        if (FunctionDefinitionRegistry.isScalarFunc(funcName) && isCompileTimeEvaluationPossible(funcExpr)) {
+
+          ScalarFunctionType scalarFunctionType = ScalarFunctionType.getScalarFunctionType(funcName);
+          switch (scalarFunctionType) {
+            case NOW:
+              funcExpr = RequestUtils.getLiteralExpression(System.currentTimeMillis());
+              break;
+            case FORMAT_DATETIME:
+              //DATETIME_FORMAT ('2020-01-01', 'yyyy-MM-dd')
+              String input = funcExpr.getFunctionCall().getOperands().get(0).getLiteral().getStringValue();
+              String format = funcExpr.getFunctionCall().getOperands().get(1).getLiteral().getStringValue();
+              long output = DateTimeFormat.forPattern(format).parseMillis(input);
+              funcExpr = RequestUtils.getLiteralExpression(output);
```

Comment:
better to implement method `eval` for each function which take operands list as argument which could be either literal or functions.
Then this logic will just be
```funcExpr = getScalarFunction(scalarFunctionType).eval(funcExpr.getFunctionCall().getOperands())```

Reply:
Most of the functions are already available in FunctionRegistry and FunctionInvoker added by @npawar. Unfortunately, those functions are in pinot-core, we can move the scalar functions into pinot-common in another PR and remove custom implementation for now and datetime format

Reply:
Added support for eval function of functions as long as all the recursive params are literal.